### PR TITLE
Add offline support

### LINKED ISSUE #10625: 建议antd 官网支持service worker
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
由于网络原因，在公司查看组件文档时经常加载不出来。故建议antd 官网支持service worker。

### What does the proposed API look like?
none

### PR DESCRIPTION
**Pull request by yesmeck**
Close #10625 

kapture 2018-06-01 at 18 42 47

### CODE REVIEW

File: site/theme/template/AppShell.jsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Empty component for app shell
+ * See https://github.com/NekR/offline-plugin/blob/master/docs/app-shell.mD
```
Comment (picodoth) [inline site/theme/template/AppShell.jsx]: app-shell.mD => app-shell.md

File: package.json [hunk 2]
```diff
@@ -135,6 +135,8 @@
     "majo": "^0.6.2",
     "mockdate": "^2.0.1",
     "moment-timezone": "^0.5.5",
+    "node-glob": "^1.2.0",
```
Comment (picodoth) [inline package.json]: we're using `require('glob)` is it the same one

Comment (picodoth) [reply package.json]: should be "glob": "^7.1.2": https://github.com/isaacs/node-glob/blob/master/package.json#L3

File: site/getExternalResources.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const glob = require('glob');
+const fs = require('fs');
+const { uniq } = require('lodash');
+
+function getExternalResources() {
+  let resources = [];
+  const files = glob.sync('{./{components,docs}/**/*.md,./site/**/*.{less,js,jsx}}');
+  files.forEach((file) => {
+    const content = fs.readFileSync(file, 'utf-8');
+    const pattern = new RegExp('(https://.+.alipayobjects.com/.+.(png|jpg|svg)|)', 'mg');
```
Comment (picodoth) [inline site/getExternalResources.js]: should it be `new RegExp('(https://.+\.alipayobjects.com/.+\.(png|jpg|svg))', 'mg');`

**Comment by picodoth** [comment]
very cool!!
🍾 🍾🍾🍾🍾🍾🍾

File: site/getExternalResources.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const glob = require('glob');
+const fs = require('fs');
+const { uniq } = require('lodash');
+
+function getExternalResources() {
+  let resources = [];
+  const files = glob.sync('{./{components,docs}/**/*.md,./site/**/*.{less,js,jsx}}');
+  files.forEach((file) => {
+    const content = fs.readFileSync(file, 'utf-8');
+    const pattern = new RegExp('(https://.+\\.alipayobjects\\.com/.+\\.(png|jpg|svg)|)', 'mg');
```
Comment (picodoth) [inline site/getExternalResources.js]: 一个转译就够了吧？

Comment (yesmeck) [reply site/getExternalResources.js]: https://github.com/eslint/eslint/issues/6148

Comment (yesmeck) [reply site/getExternalResources.js]: https://github.com/eslint/eslint/issues/7656

**Review by picodoth — APPROVED** [review]

**Comment by paranoidjk** [comment]
http://ant.design/  doesn't register a service worker in my chrome right now. Looks like have a bug ?

**Comment by zombieJ** [comment]
@paranoidjk , try https

**Comment by yesmeck** [comment]
已开启 force https

**Pull request merged**